spyder/plugins/editor/widgets/base.py

```diff
@@ -278,46 +278,111 @@ def unhighlight_current_cell(self):
         """Unhighlight current cell"""
         self.clear_extra_selections('current_cell')
 
-    #------Brace matching
-    def find_brace_match(self, position, brace, forward):
+    def in_comment(self, cursor=None, position=None):
+        """Returns true if the given position is inside a comment
+
+        Trivial default implementation. To be overridden by subclass.
+        This function is used to define the default behaciour of
+        self.find_brace_match.
+        """
+        return False
+
+    def in_string(self, cursor=None, position=None):
+        """Returns true if the given position is inside a string
+
+        Trivial default implementation. To be overridden by subclass.
+        This function is used to define the default behaciour of
+        self.find_brace_match.
+        """
+        return False
+
+    def find_brace_match(self, position, brace, forward,
+                         ignore_brace=None, stop=None):
+        """Returns position of matching brace
+
+        Parameters
+        ----------
+        position : int
+            The position of the brace to be matched.
+        brace : {'[', ']', '(', ')', '{', '}'}
+            The brace character to be matched.
+            [ <-> ], ( <-> ), { <-> }
+        forward : boolean
+            Whether to search forwards or backwards for a match.
+        ignore_brace : callable taking int returning boolean, optional
+            Whether to ignore a brace (as function of position).
+        stop : callable taking int returning boolean, optional
+            Whether to stop the search early (as function of position).
+
+        If both ignore_brace and brace are None, the default behaviour
+        then brace matching is handled differently depending on whether
+        the starting position cursor is inside a string, comment or
+        regular code. The member functions self.in_comment and
+        self.in_string are used.
+
+        If exactly one of ignore_brace and brace is None, it is
+        replaced by function returning False for every position. I.e.:
+            lambda pos: False
+
+        Returns
+        -------
+        The position of the matching brace. If no matching brace
+        exists, then None is returned.
+        """
+
+        if ignore_brace is None and stop is None:
+            if self.in_string(position=position):
+                # Only search inside the current string
+                def stop(pos): return not self.in_string(position=pos)
+            elif self.in_comment(position=position):
+                # Only search inside the current comment
+                def stop(pos): return not self.in_comment(position=pos)
+            else:
+                # Ignore braces inside strings and comments
+                def ignore_brace(pos): return self.in_string(position=pos) \
+                    or self.in_comment(position=pos)

```

Comment:
Please reformat this function as follows:

```python
def ignore_brace(pos):
    return (self.in_string(position=pos) or
            or self.in_comment(position=pos))
```

We don't accustom to use continuation lines anymore.